app/controllers/foreman_tasks/api/recurring_logics_controller.rb

```diff
@@ -38,10 +38,22 @@ def cancel
       def resource_class
         ForemanTasks::RecurringLogic
       end
+                      
+      api :POST, '/recurring_logics/bulk/destroy', N_('Delete recurring logics by search query')
+      param :search, String, :desc => N_('Search query'), :required => true
+      def bulk_destroy
+        scope = resource_scope.search_for(params[:search])
+        destroyed = scope.destroy_all
+        if destroyed
+          render json: { destroyed: destroyed }
+        else
+          render json: {}, status: :bad_request
```

Comment:
I haven't tested the errors yet, but would we want to return some sort of error message?

Reply:
What kind of error message? `recurring logics could not be destroyed`?

Reply:
The message that would tells us what went wrong and why this is a bad request. Returning error status with empty response does not provide any hints.

Reply:
added the error if destroy fails